CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+cmake_minimum_required(VERSION 3.10.2)
+
+# where to find our CMake modules
+set(CMAKE_MODULE_PATH "$ENV{CARMA_OPT_DIR}/cmake")
+# Required for std::lock and syntax
+set(CMAKE_CXX_STANDARD 17)
+
+# Needed so CMake can find the vendored PROJ4 module file. Th FindPROJ4.cmake
+# module file can be removed if we upgrade to a more recent PROJ version. See
+# https://github.com/usdot-fhwa-stol/carma-platform/issues/2139 for updates.
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+project(carma-geodetic
+    DESCRIPTION "CARMA geodetic library"
+    HOMEPAGE_URL https://github.com/usdot-fhwa-stol/carma-geodetic-lib
+	VERSION 0.0.1
+    LANGUAGES CXX
+)
+
+find_package(Threads)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
+include(cmake/dependencies.cmake)
+
+add_library(${PROJECT_NAME} SHARED)
+target_link_libraries( ${PROJECT_NAME}
+ ${PROJ4_LIBRARIES}  # Note: Newer versions of PROJ use PROJ::proj
+  GSL  # Note: Newer versions of GSL use Microsoft.GSL::GSL
+  units::units
+)
+target_include_directories(${PROJECT_NAME} 
+    PUBLIC
+        $<INSTALL_INTERFACE:include>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/src
+)
+target_link_libraries( ${PROJECT_NAME}
+ ${PROJ4_LIBRARIES}  # Note: Newer versions of PROJ use PROJ::proj
+  GSL  # Note: Newer versions of GSL use Microsoft.GSL::GSL
+  units::units
+)
+set_target_properties(${PROJECT_NAME} PROPERTIES
+    SOVERSION 1
+)
```

Comment:
I don't think this is needed.

Reply:
It looks like to me this is SOVERSION is used by several things in the CPack functionality.
https://github.com/usdot-fhwa-stol/carma-builds/blob/cd6e4ce3d443539ec4fd598d1c41225d79ecdcc4/cmake/Packing.cmake#L19

Reply:
I see. In that case, I believe this should be `0` since the library is on version `0.0.1`. However, I'm not super familiar with soversion conventions and packaging.

Here's a link to a presentation Craig Scott made that has some relevant info (starting at slide 34): [link](https://github.com/CppCon/CppCon2019/blob/master/Presentations/deep_cmake_for_library_authors/deep_cmake_for_library_authors__craig_scott__cppcon_2019.pdf)

Reply:
I don't think you can set this to `0`. When I set it to `0` I get errors from the Packing.cmake that indicate it is not set at all 
```
CMake Error at /opt/carma/cmake/Packing.cmake:19 (message):
  Target property SOVERSION must be defined for shared library packages.
```

This kind of makes sense since I have never seen a shared library file/simlink by <library_name>.so.0 but I could be missing something here.

Reply:
That appears to be a bug in the `Packing.cmake` file. The CMake documentation mentions [`get_target_property()`](https://cmake.org/cmake/help/latest/command/get_target_property.html) will set `<VAR>` to `<VAR>-NOTFOUND` or an empty string, depending on the context. The `if` state should be changed to something like this:

```cmake
if(NOT "${so_version}")
  # ...
endif()
```
I don't think I got the syntax quite right, but this shows the idea.

I don't see anywhere in the [CMake documentation](https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html) preventing `SOVERSION` from being `0`, and I've seen examples of some projects having a `0` soversion.

Reply:
Ok let me check this out and see if I can fix it.

Reply:
Addressed in PR https://github.com/usdot-fhwa-stol/carma-builds/pull/26